File: fintechplatform/src/main/java/com/fintechplatform/android/enterprise/address/ui/EnterpriseAddressUI.java
package com.fintechplatform.android.enterprise.address.ui;

import android.content.Context;
import android.content.Intent;

import com.android.volley.toolbox.Volley;
import com.fintechplatform.android.api.NetModule;
import com.fintechplatform.android.enterprise.api.EnterpriseAPIModule;
import com.fintechplatform.android.models.DataAccount;

public class EnterpriseAddressUI {

    private String hostName;
    private DataAccount configuration;

    protected static EnterpriseAddressUI instance;

    public EnterpriseAddressUI(String hostName, DataAccount configuration) {
        this.hostName = hostName;
        this.configuration = configuration;
    }

    protected EnterpriseAddressViewComponent buildAddressComponent(Context context, EnterpriseAddressContract.View view) {
        return DaggerEnterpriseAddressViewComponent.builder()
                .netModule(new NetModule(Volley.newRequestQueue(context),instance.hostName))
                .enterpriseAPIModule(new EnterpriseAPIModule(instance.hostName))
                .enterpriseAddressPresenterModule(new EnterpriseAddressPresenterModule(view, instance.configuration))
                .build();

    }

    public EnterpriseAddressViewComponent createAddressComponent(Context context, EnterpriseAddressContract.View view){
        return instance.buildAddressComponent(context, view);
    }

    public void start(Context context) {
        instance = this;
        Intent intent = new Intent( context, EnterpriseAddressActivity.class);
        context.startActivity(intent);
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/enterprise/address/EnterpriseAddressUI.kt
package com.fintechplatform.ui.enterprise.address

import android.content.Context
import android.content.Intent
import android.os.Bundle
import com.android.volley.toolbox.Volley
import com.fintechplatform.api.enterprise.api.EnterpriseAPIModule
import com.fintechplatform.api.net.NetModule
import com.fintechplatform.ui.enterprise.address.di.DaggerEnterpriseAddressViewComponent
import com.fintechplatform.ui.enterprise.address.di.EnterpriseAddressPresenterModule
import com.fintechplatform.ui.enterprise.address.di.EnterpriseAddressViewComponent
import com.fintechplatform.ui.models.DataAccount


class EnterpriseAddressUI(private val hostName: String, private val dataAccount: DataAccount) {

    fun startActivity(context: Context) {
        val intent = Intent(context, EnterpriseAddressActivity::class.java)
        val args = Bundle()
        args.putString("hostname", hostName)
        args.putParcelable("dataAccount", dataAccount)
        intent.putExtras(args)
        context.startActivity(intent)
    }

    fun createFragment(): EnterpriseAddressFragment {
        return EnterpriseAddressFragment.newInstance(hostName, dataAccount)
    }

    object Builder {
        fun buildAddressComponent(context: Context, view: EnterpriseAddressContract.View, hostName: String, dataAccount: DataAccount): EnterpriseAddressViewComponent {
            return DaggerEnterpriseAddressViewComponent.builder()
                    .netModule(NetModule(Volley.newRequestQueue(context), hostName))
                    .enterpriseAPIModule(EnterpriseAPIModule(hostName))
                    .enterpriseAddressPresenterModule(EnterpriseAddressPresenterModule(view, dataAccount))
                    .build()

        }
    }
}
